### ISSUE TITLE
Wrong detection of getInitialProps on _app.tsx

### ISSUE THREAD

**Issue by aganjali** [aralroca/next-translate#486]
Hi everyone

seems that the lib detect getInitialProps on _app.js (or any other pages) with definition like below:

```
import { Provider } from 'react-redux' 
import store from '@store'

import type { AppProps } from 'next/app'
import '@styles/index.scss'

const MyApp: React.FC<AppProps> = ({ Component, pageProps }) => {
  

  return (
    <Provider store={store}>
      <Component {...pageProps} />
    </Provider>
  )
}

export default MyApp
```
with this definition model i get Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`

and on the other pages than _app.js for example pages/index.tsx 
```

import Head from 'next/head' 
import Link from 'next/link' 
interface Props {}

const Home: React.FC<Props> = () => {
   
  return (
    <div className="absolute inset-0 w-full h-full ">
      <Head>
        <title>Create Next App</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      
      >
        change from {test}
      </button>
      <Link href="/test">
        <a>test</a>
      </Link>
    </div>
  )
}

export default Home

```
i get this warning:

 [next-translate] In Next 10.x.x there is an issue related to i18n and getInitialProps. We recommend to replace getInitialProps to getServerSideProps on /index.tsx. Issue: https://github.com/vercel/next.js/issues/18396

UPDATE: When i remove those React.FC<Props> the errors gone!

UPDATE: With code below also i get the warning  [next-translate] In Next 10.x.x there is an issue related to i18n and getInitialProps. We recommend to replace getInitialProps to getServerSideProps on /index.tsx. Issue: https://github.com/vercel/next.js/issues/18396

```
import Head from 'next/head';

const Home = () => {
  return (
    <div className="flex-1 w-full  ">
      <Head>
        <title>Test</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
    </div>
  );
};
Home.getLayout = (page: React.ReactNode) => {
  <div className="absolute flex flex-col inset-0">{page}</div>;
};

export default Home;

```

**Comment by aganjali** [comment]
forgot to mention

  "dependencies": { 
    "next": "10.0.5", 
    "next-translate": "^1.0.2", 
    "react": "17.0.1",
    "react-dom": "17.0.1", 
  },

**Comment by Divlo** [comment]
Indeed I have the same issue, it might be a duplicate of this issue : #453.

**Comment by Shariaty** [comment]
I am getting this warning when getting a production built of my app:

`Warning: You have opted-out of Automatic Static Optimization due to getInitialProps in pages/_app. This does not opt-out pages with getStaticProps`

While, there is no getInitialProps whatsoever anywhere in my app, I found out that it happens as soon as I make use of nextTranslate in my next.config.js, Also, it turns all my pages even those which are fully static into server side rendered!!?

**NextJs Version: 10.0.9
next-translate Version: 1.0.4**

**Comment by BjoernRave** [comment]
for the guys having problems with this I can recommend to stay on `1.0.1`. For me everything is still fine there

**Comment by AndysonDK** [comment]
This really needs to be fixed! @Shariaty what's your workaround?

**Comment by jahirfiquitiva** [comment]
I'm getting the same issue with the latest release 😭

**Comment by aralroca** [comment]
I believe this issue has been resolved with Next-translate 2.0, as we have rewritten the plugin and replaced the use of regex with a parser. This update has automatically addressed many of the issues related to attempting to parse items with regex.

I trust that this solution has fully resolved the matter, and as such, I will proceed to close the issue. Additionally, I would like to inform you that we have moved the plugin to this repository: https://github.com/aralroca/next-translate-plugin.

If, for any reason, you find that the issue has not been fully resolved, please feel free to reopen the issue on the next-translate-plugin repository.

Thank you very much.
